examples/AffinePlaneCurve.jl

```diff
@@ -221,13 +227,13 @@ end
 # gives the common components of two affine plane curves
 
 @doc Markdown.doc"""
-common_components(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem(C::AffinePlaneCurve)
-> Given two affine plane curves C and D, returns the affine plane curve consisting of the common components, or an error if they do not have a common component.
+common_component(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem(C::AffinePlaneCurve)
+> Returns the affine plane curve consisting of the common component of C and D, or an empty array if they do not have a common component.
 """
-function common_components(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
+function common_component(C::AffinePlaneCurve{S}, D::AffinePlaneCurve{S}) where S <: FieldElem
   G = gcd(C.eq, D.eq)
   if G == 1
```

Comment:
```suggestion
  if isone(G)
```
Otherwise you are comparing an mpoly to an integer (which may work, but my guess is that it might be less efficient).

A similar change could be made in a bunch of other places; also more places could be using `isempty`.

Just to stress this: I don't think these comments warrant holding up this PR; I just mention them so @delphinepol is aware and can think about changing this at some point

Reply:
Thank you very much for all the comments !